src/sentry/api/endpoints/project_metrics_extraction_rules.py

```diff
@@ -85,11 +86,33 @@ def delete(self, request: Request, project: Project) -> Response:
 
         with handle_exceptions():
             with transaction.atomic(router.db_for_write(SpanAttributeExtractionRuleConfig)):
-                delete_extraction_rule_config(project, config_update)
-                schedule_invalidate_project_config(
-                    project_id=project.id, trigger="span_attribute_extraction_configs"
-                )
                 for config in config_update:
+                    try:
+                        span_attribute = SpanAttributeExtractionRuleConfig.objects.get(
+                            project=project, span_attribute=config["spanAttribute"]
+                        )
+                    except SpanAttributeExtractionRuleConfig.DoesNotExist:
+                        # ignore non existing extraction rules
+                        logger.info(
+                            "Extraction rule does not exist while trying to delete",
+                            extra={"config": config},
+                        )
+                        continue
+
+                    config_mris = []
+                    for condition in span_attribute.conditions.all():
+                        config_mris.extend(condition.generate_mris())
+
+                    span_attribute.delete()
+                    # delete connected alert rules and dashboards widgets to an extraction rule
+                    AlertRule.objects.get_for_metrics(project.organization, config_mris).delete()
```

Comment:
When an alert rule is deleted, there's multiple dependent actions that need to be taken - see https://github.com/getsentry/sentry/blob/21aefaf841727b1f6929b6627e4ad1fccc46b82f/src/sentry/incidents/logic.py#L1005 - is there a reason we're not using this function here?